test/ApplicationTest.php

```diff
@@ -268,6 +268,17 @@ public function testThrowsExceptionWhenDispatchingUncallableMiddleware($middlewa
         });
     }
 
+    public function testExecuteMiddlewareIfNotRouteMatch()
```

Comment:
Note to self: rename this to `testDispatchMiddlewareExecutesNextMiddlewareInStackIfNoRouteMatchPresent`.